libcalico-go/lib/backend/k8s/resources/ipam_config.go

```diff
@@ -79,10 +80,62 @@ func (c ipamConfigClient) toV1(kvpv3 *model.KVPair) (*model.KVPair, error) {
 	}, nil
 }
 
-// toV3 takes the given v1 KVPair and converts it into a v3 representation, suitable
+// There's two possible kV formats to be passed to backend ipamConfig.
+// 1. Libcalico-go IPAM passes a v1 model.IPAMConfig directly. [libcalico-go/lib/ipam/ipam.go]
+// 2. Calico-apiserver storage passes a kv with libapiv3.IPAMConfig
+
+// isV1KVP return if the KV value is in v1 format.
+func isV1KVP(kvpv1 *model.KVPair) bool {
+	switch kvpv1.Value.(type) {
+	case *model.IPAMConfig:
+		return true
+	case *libapiv3.IPAMConfig:
+		return false
+	default:
+		log.Panic("ipamConfigClient : wrong value interface type")
+	}
+	return false
+}
+
+// isV1Key return if the Key is in v1 format.
+func isV1Key(key model.Key) bool {
+	switch key.(type) {
+	case model.IPAMConfigKey: // used by Calico IPAM [libcalico-go/lib/ipam/ipam.go]
+		return true
+	case model.ResourceKey: // used by clientv3 resource API [libcalico-go/lib/clientv3/resources.go]
+		return false
+	default:
+		log.Panic("ipamConfigClient : wrong key interface type")
+	}
+	return false
+}
+
+// For the first point, toV3 takes the given v1 KVPair and converts it into a v3 representation, suitable
 // for writing as a CRD to the Kubernetes API.
+//
+// Also note the name of the resource are hard coded to "default".
```

Comment:
This comment / function is a little different now, because it's possible that it will receive a v3 version of the object on input, right?

Should we just short-circuit if given an already assembled v3 object?